db/db_impl/db_impl_files.cc

```diff
@@ -58,7 +58,7 @@ Status DBImpl::EnableFileDeletions(bool force) {
   // Job id == 0 means that this is not our background process, but rather
   // user thread
   JobContext job_context(0);
-  bool file_deletion_enabled = false;
+  int counter;
```

Comment:
I know it does not matter, but can you either initialize this to 0 or put a comment that says it is initialized in the lock block.  My natural reaction was to think it was not guaranteed to be initialized.

Reply:
Because we compile with warnings as errors, not initializing it enforces that it be set on all paths (leading to use). I'll add a comment.